target1: [-0.6, -0.2, 0.35]
target2: [-0.55, 0, 0.45]
target3: [-0.5, 0.2, 0.35]